src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs

```diff
@@ -316,15 +316,13 @@ private static string GetExpectedRedirectUri(string clientId)
         /// <param name="authenticationRequestParameters"></param>
         /// <param name="logger"></param>
         /// <exception cref="MsalClientException"></exception>
-        public static void ValidateAuthParams(
+        private static void ValidateAuthParams(
             AuthenticationRequestParameters authenticationRequestParameters,
             ILoggerAdapter logger)
         {
             //MSAL Runtime throws an ApiContractViolation Exception with Tag: 0x2039c1cb (InvalidArg)
             //When no scopes are passed, this will check if user is passing scopes
-            var scopes = authenticationRequestParameters.Scope?.ElementAtOrDefault(0);
-
-            if (string.IsNullOrWhiteSpace(scopes))
+            if (!ScopeHelper.HasNonMsalScopes(authenticationRequestParameters.Scope))                
```

Comment:
@gladjohn - since MSAL C++ has undefined behavior when using one of the 3 reserved scopes, let's throw an error message here.